### DISCUSSION TITLE
SSL_CTX_set_tmp_rsa_callback

### DISCUSSION THREAD

**Discussion by nagu1137** [openssl/openssl#23769] [Q&A]
SSL_CTX_set_tmp_rsa_callback is this functon deprecated in 3.1 version?
we are not getting any deprecated warning in the logs but by looking at .h it is deprecated.

**Comment by davidben** [comment]
I suspect it being a macro prevents it from showing deprecation warnings. However, keep in mind that this function has done absolutely nothing since OpenSSL 1.1.0, so, deprecation warning or no, you should simply delete it from your application. This callback is a remnant of the old, weak RSA export ciphers, which have not been relevant for a couple decades.

**Comment by nagu1137** [comment]
thanks for responding.

Do we have any alternative for this?

-Thanks and Regards,

*N.Nagaraju*

**Reply by davidben** [reply]
The alternative is to not call the function. Simply delete the function call. Your application will not change in behavior. _The function does not do anything_.
